pkg/river/vm/vm.go

```diff
@@ -173,26 +173,59 @@ func (vm *Evaluator) evaluateBlockOrBody(scope *Scope, assoc map[value.Value]ast
 			continue
 
 		case tf.IsAttr() && len(blocks) > 0:
-			return fmt.Errorf("%q must be an attribute, but is used as a block", fullName)
+			return diag.Diagnostic{
+				Severity: diag.SeverityLevelError,
+				StartPos: ast.StartPos(blocks[0]).Position(),
+				EndPos:   ast.EndPos(blocks[len(blocks)-1]).Position(),
```

Comment:
The problem here is that the highlight range might include blocks which are fine. Imagine this scenario:

```river 
invalid_block_name { ... } 
valid_block_name { ... } 
invalid_block_name { ... } 
```

The error will report the entire range from the first and last block as invalid, even though the block in the middle is fine.

Rather than reporting one diagnostic for the entire set of blocks, you can return `diag.Diagnostics` (a slice of `diag.Diagnostic`) and add a diagnostic for each invalid block in `blocks`.